src/dpq2/conv/from_d_types.d

```diff
@@ -37,8 +51,41 @@ Value toValue(T : bool)(T v) @trusted
     return Value(buf, detectOidTypeFromNative!T, false, ValueFormat.BINARY);
 }
 
+/// Constructs Value from Date
+Value toValue(T)(T v)
+if (is(Unqual!T == Date))
+{
+    auto days = cast(int)(v - POSTGRES_EPOCH_DATE).total!"days";
+    return Value(nativeToBigEndian(days).dup, OidType.Date, false);
+}
+
+/// Constructs Value from TimeOfDay
+Value toValue(T)(T v)
+if (is(Unqual!T == TimeOfDay))
+{
+    long ms = (v.second + v.minute*60L + v.hour*3_600L)*1_000_000;
+    return Value(nativeToBigEndian(ms).dup, OidType.Time, false);
+}
+
+/// Constructs Value from TimeStampWithoutTZ
+Value toValue(T)(T v)
+if (is(Unqual!T == TimeStampWithoutTZ))
+{
+    return toValue(cast(SysTime)v); // works similarly to SysTime, but TZ is not used for conversion
```

Comment:
Is this cast really works?

Reply:
It works because of added `opCast` in `TimeStampWithoutTZ`.
I've added tests there and also static method to construct it from `SysTime` which can be helpfull.

Reply:
I am not familiar with `opCast`... Probably, stupid question: If anyone accidentally removes opCast from TimeStampWithoutTZ struct this casting will start to return garbage?

I.e., why not to use toSysTime method instead of opCast here?

Reply:
Without opCast in place it wont compile, so no need to worry.
As I've removed it, there is no cast to SysTime anymore.